crypto/src/main/java/org/springframework/security/crypto/factory/PasswordEncoderFactories.java

```diff
@@ -71,7 +73,8 @@ public static PasswordEncoder createDelegatingPasswordEncoder() {
 		encoders.put("MD4", new org.springframework.security.crypto.password.Md4PasswordEncoder());
 		encoders.put("MD5", new org.springframework.security.crypto.password.MessageDigestPasswordEncoder("MD5"));
 		encoders.put("noop", org.springframework.security.crypto.password.NoOpPasswordEncoder.getInstance());
-		encoders.put("pbkdf2", new Pbkdf2PasswordEncoder());
+		encoders.put("pbkdf2", Pbkdf2PasswordEncoder.defaultsForSpringSecurity_v5_5());
+		encoders.put("pbkdf2-sha256", Pbkdf2PasswordEncoder.defaultsForSpringSecurity_v5_8());
```

Comment:
I think the -sha256 should be also related to v5_8 or all alg parameters since there are other parameters that could change. I realize that the recommendations are based on the algorithm, but these recommendations are for now and may change. That would make it difficult for us to provide additional settings later.

My quick thought is that I'd prefer the id to contain all parameters even if it is verbose as this is easier for users to understand and will be similar to other encodings work (I.e. [phc](https://github.com/spring-projects/spring-security/issues/9833), bcrypt, etc)

Reply:
> My quick thought is that I'd prefer the id to contain all parameters even if it is verbose

I still think it would be too verbose and not that easy to understand.

I decided to align the `encodingId` to the static factory method name, for example:

- `argon2@SpringSecurity_v5_8` -> `Argon2PasswordEncoder.defaultsForSpringSecurity_v5_8()`
- `scrypt@SpringSecurity_v5_8` -> `SCryptPasswordEncoder.defaultsForSpringSecurity_v5_8()`
- `pbkdf2@SpringSecurity_v5_8` -> `Pbkdf2PasswordEncoder.defaultsForSpringSecurity_v5_8()`